docs/templates.md

```diff
@@ -0,0 +1,527 @@
+
+#### Limiting the number of posts listed
+
+The "limit" attribute restricts the number of posts that come back in the
+current page. Note that if you haven't set the page, changing this is equivalent
+to changing "num". If you _have_ set the page, then "limit" limits how many posts
+of that page are returned. Defaults to 20.
+
+##### Example 
+
+```
+<wpPosts limit="10"">
+  <wpTitle />
+</wpPosts>
+```
+
+This will display only 10 posts.
+
+#### Offset 
+
+The "offset" attribute affects how many posts are skipped in the current page.
+If you don't set the page, then this is just the number of posts that are
+skipped before the first posts are returned. If you have set the page, then the
+posts will skip to the current page, then an additional number of posts will be
+skipped using the "offset".
+
+##### Example 
+
+```
+<wpPosts page="3" num="10" offset="5">
+  <wpTitle />
+</wpPosts>
+```
+
+This will skip 20 posts (to get to the third page), then skip five more posts
+for the offset, then display the next 10 posts.
+
+### `<wpPostByPermalink>`
+
+This tag will retrieve the post that corresponds to the current request URL. 
+
+#### Example
+
+Say this post exists in WordPress:
+
+```
+Title: A great post
+Slug: a-great-post
+
+Body: Wow, this is an excellent post.
+
+```
+
+A visitor goes to the URL: `https://myblog.com/2016/10/great-post`. The template the server uses is `post.tpl` and it looks like this:
+
+```
+<wpPostByPermalink>
+  <h3><wpTitle /><h3>
+  
+  <main>
+    <wpContent />
+  </main>
+</wpPostByPermalink>
+```
+
+That will result in this rendered HTML:
+
+```
+<h3>A great post</h3>
+
+<main>
+  Wow, this is an excellent post.
+</main>
+```
+
+### `<wp>`
+
+Wrap your templates in the `wp` tag so that multiple post lookups can happen at
+the same time -- this will speed up rendering.
+
+#### Example
+
+This makes three calls to the WordPress API:
+
+```
+<wpPosts tag="featured"></wpPosts>
+
+<wpPosts tag="secondary"></wpPosts>
+
+<wpPosts limit="10"></wpPosts>
+```
+
+This makes only one call to the API, so is much faster:
+
+```
+<wp>
+  <wpPosts tag="featured"></wpPosts>
+
+  <wpPosts tag="secondary"></wpPosts>
+
+  <wpPosts limit="10"></wpPosts>
+</wp>
+
+```
+
+### `<wpNoPostDuplicates />`
+
+Insert this tag before a bunch of `wpPosts` tags to make sure that no duplicates
+are shown on the rendered page. You can use it so that multiple post listings 
+don't show the same posts.
+
+#### Example
+
+```
+<wp>
+  <div id="featured">
+    <wpPosts tag="featured"><wpTitle /></wpPosts>
+  </div>
+
+  <div id="secondary">
+    <wpPosts limit="10"><wpTitle /></wpPosts>
+  </div>
+</wp>
+```
+
+If there are three posts, "Post 1", "Post 2", and "Post 3", and "Post 1" is a
+featured post, then rendering this template will result in:
+
+```
+<div id="featured">
+  Post 1
+</div>
+
+<div id="secondary">
+  Post 1
+  Post 2
+  Post 3
+</div>  
+```
+
+Post 1 appears twice! If you add `<wpNoPostDuplicates />`:
```

Comment:
I misread this on first parse as "Post 1 appears twice if you add `wpPostNoDuplicates`. Some whitespace would help and maybe a "but"